src/Symfony/Component/HttpFoundation/File/UploadedFile.php

```diff
@@ -236,6 +236,24 @@ public function move($directory, $name = null)
         throw new FileException($this->getErrorMessage());
     }
 
+    public function moveAndRename(string $targetDirectory, string $fileNamePattern)
+    {
+        $fileName = strtr($fileNamePattern, [
+            '[contenthash]' => sha1_file($this->getRealPath()),,
```

Comment:
i think we should prefer a reserved character, e.g. `%key%`

Reply:
One comma is enough here